sycl/test/scheduler/LinkedAlloca.cpp

```diff
@@ -0,0 +1,85 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -I %sycl_source_dir %s -o %t.out
+// RUN: env SYCL_PRINT_EXECUTION_GRAPH=always %t.out
+// RUN: cat graph_7after_addHostAccessor.dot
+// RUN: cat graph_7after_addHostAccessor.dot | FileCheck %s
+
+#include <CL/sycl.hpp>
+
+namespace S = cl::sycl;
+
+void test() {
+  auto EH = [](S::exception_list EL) {
+    for (const std::exception_ptr &E : EL) {
+      throw E;
+    }
+  };
+
+  S::queue Queue(EH);
+
+  static const size_t BSIZE = 10;
+  S::buffer<int, 1> Buf{BSIZE};
+
+  // 0. submit kernel
+  Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::global_buffer>
+        GeneratorAcc(Buf, CGH);
+
+    auto GeneratorKernel = [GeneratorAcc]() {
+      for (size_t Idx = 0; Idx < GeneratorAcc.get_count(); ++Idx)
+        GeneratorAcc[Idx] = BSIZE - Idx;
+    };
+
+    CGH.single_task<class GeneratorTask0>(GeneratorKernel);
+  });
+  // 1. create host-accessor
+  {
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::host_buffer>
+        Acc(Buf);
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx)
+      Acc[Idx] = -1;
+  }
+
+  // 2. submit kernel
+  Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::global_buffer>
+        GeneratorAcc(Buf, CGH);
+
+    auto GeneratorKernel = [GeneratorAcc]() {
```

Comment:
Can we have a googletest/unittest instead and just check if a linked alloca is in the dependency list of a new alloca? Examples:
https://github.com/intel/llvm/blob/sycl/sycl/unittests/scheduler/BlockedCommands.cpp
https://github.com/intel/llvm/blob/sycl/sycl/unittests/scheduler/FinishedCmdCleanup.cpp

Reply:
There is a unit-test. It's added within this PR. See https://github.com/intel/llvm/pull/1470/files#diff-d96769fc46c97d667b6eb4625b44970dR1.

Reply:
Great! I didn't get to it.
So, why do we need LinkedAlloca.cpp test? Can we have LinkedAllocaDependencies.cpp  ?

Reply:
Repeating question to make it clear:
Why do we need LinkedAlloca.cpp test? Isn't LinkedAllocaDependencies.cpp enough?

Reply:
`sycl/test/scheduler/LinkedAlloca.cpp` isn't needed as it only looks at the graph and is not really checking data structures.